### DISCUSSION TITLE
Xiaomi WXKG01LM - Won't stay connected

### DISCUSSION THREAD

**Discussion by flipsidelive** [Koenkk/zigbee2mqtt#15443] [Devices]
Hi Everyone,

I'm struggling with the Xiaomi WXKG01LM button. I bought several of these buttons as they are small and relatively cheap. I get them to connect no problem (with the strange hold reset and then click the button lots of times method) and I'm able to use them for a period of an hour or so but after that, they stop working. Initially, I tried ZHA but after no joy so moved to Zigbee2MQTT.

I'm using an ITead Sonoff Zigbee 3.0 USB Dongle Plus V2 model "ZBDongle-E" adapter which I know is listed as experimental but work so well for everything else. I have no repeaters and only 6 other devices. I've checked the batteries and all looks good. Its reporting a strange power outage count of 751.

I'm at a loss. Anyone got any tips?

Thanks

**Comment by flipsidelive** [comment]
Anyone come across this type of problem?

**Reply by d-allin** [reply]
I can confirm this exact same combination of adapter and button on my setup has the same issue.
This occurred in ZHA and I moved to Zigbee2MQTT in hopes of a better outcome.
In my case, they seem to "drop off" after hours (usually 5+ hours) of not being used. When I re-pair them, the Power_outage_count updates (latest from 220 to 266).

**Reply by d-allin** [reply]
Giving this guy's solution a shot. First impressions are good; staying connected in map view (the lines would disappear before).
https://old.reddit.com/r/homeassistant/comments/x5n21j/xiaomi_wxkg01lm_keep_disconnet_from_zigbee2mqtt/j0cw8pl/

**Reply by flipsidelive** [reply]
This looks very promising. I'll take a look. Thankyou very much for taking the time to reply.

**Reply by d-allin** [reply]
The behaviour I observed before was that `[Device] action changed to None`, then I would have to re-pair the device before any new button presses were seen.

After flashing ncp-uart-sw_679_115200.gbl + nvm3_initfile.gbl, the device continues to respond after action changing to None. I'm at a record 14 hours of solid connection.

Let me know how this goes for you.

**Reply by pboschi** [reply]
I'm trying ncp-uart-sw_v6.10.3_115200.gbl, which is the latest... any reason you didn't try it first? It's the same version of the original, but maybe a different build...

**Reply by d-allin** [reply]
No, I did not have a particular reason for using ncp-uart-sw_679_115200.gbl other than blindly following the reddit post.

**Reply by pboschi** [reply]
> No, I did not have a particular reason for using ncp-uart-sw_679_115200.gbl other than blindly following the reddit post.

well, you saved time, because with the latest version the button still disconnects after some hours.

**Reply by d-allin** [reply]
I've tried both `ncp-uart-sw_655_115200.gbl` and `ncp-uart-sw_679_115200.gbl` with success. (The original poster on reddit mentions 679 in a separate post.)
Don't forget to flash `nvm3_initfile.gbl` as well.

**Reply by Merkalov** [reply]
> I've tried both `ncp-uart-sw_655_115200.gbl` and `ncp-uart-sw_679_115200.gbl` with success. (The original poster on reddit mentions 679 in a separate post.) Don't forget to flash `nvm3_initfile.gbl` as well.

Tell me please, is there an instruction for updating the firmware for stupid people?) My xiaomi switches just the same disconnect.

**Reply by d-allin** [reply]
> > I've tried both `ncp-uart-sw_655_115200.gbl` and `ncp-uart-sw_679_115200.gbl` with success. (The original poster on reddit mentions 679 in a separate post.) Don't forget to flash `nvm3_initfile.gbl` as well.
> 
> Tell me please, is there an instruction for updating the firmware for stupid people?) My xiaomi switches just the same disconnect.

The instructions are a bit spread out, but basically:

1. Download ncp-uart-sw_655_115200.gbl (or ncp-uart-sw_679_115200.gbl) and nvm3_initfile.gbl from https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP
2. Follow the "Firmware Flashing" instructions in the manual (https://sonoff.tech/wp-content/uploads/2022/07/SONOFF-Zigbee-3.0-USB-dongle-plus-firmware-flashing-.pdf) to flash *_115200.gbl and nvm3_initfile.gbl to the device.

Notes:

- I used ExtraPuTTY to do the flashing, using the instructions above.
- The "Elelabs EZSP Firmware Update Utility" mentioned in the git project did not work for me, but using ExtraPuTTY and following the PDF instructions did.
- You'll need to remove the casing from the device to access the buttons referenced in the PDF guide (see https://i0.wp.com/itead.cc/wp-content/uploads/2021/03/1000pxDongle-E-1000px-06.jpg).

**Reply by Merkalov** [reply]
> > > I've tried both `ncp-uart-sw_655_115200.gbl` and `ncp-uart-sw_679_115200.gbl` with success. (The original poster on reddit mentions 679 in a separate post.) Don't forget to flash `nvm3_initfile.gbl` as well.
> > 
> > 
> > Tell me please, is there an instruction for updating the firmware for stupid people?) My xiaomi switches just the same disconnect.
> 
> The instructions are a bit spread out, but basically:
> 
> 1. Download ncp-uart-sw_655_115200.gbl (or ncp-uart-sw_679_115200.gbl) and nvm3_initfile.gbl from https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP
> 2. Follow the "Firmware Flashing" instructions in the manual (https://sonoff.tech/wp-content/uploads/2022/07/SONOFF-Zigbee-3.0-USB-dongle-plus-firmware-flashing-.pdf) to flash *_115200.gbl and nvm3_initfile.gbl to the device.
> 
> Notes:
> 
> * I used ExtraPuTTY to do the flashing, using the instructions above.
> * The "Elelabs EZSP Firmware Update Utility" mentioned in the git project did not work for me, but using ExtraPuTTY and following the PDF instructions did.
> * You'll need to remove the casing from the device to access the buttons referenced in the PDF guide (see https://i0.wp.com/itead.cc/wp-content/uploads/2021/03/1000pxDongle-E-1000px-06.jpg).

Thanks, it works!
Uploaded instructions to youtube:
https://www.youtube.com/watch?v=CympfXuwtyQ

**Reply by beetlez** [reply]
Thanks! I was almost giving up and shopping for a new Zigbee dongle.
I have the exact combo, ITead Sonoff Zigbee 3.0 USB Dongle Plus V2 model "ZBDongle-E" adapter with some old Xiaomi buttons WXKG01LM. Symptoms exactly same per described above.

After flashing with ncp-uart-sw_679_115200.gbl & nvm3_initfile.gbl (I used the python scripts did not need to unscrew/dismantle the dongle), it works like a charm!! 

Thanks for the solving my 1 week long issue (caused me a few hours of sleep for that week!)

**Reply by sbakopanos** [reply]
For anyone looking for a painless way to flash the firmware:

https://darkxst.github.io/silabs-firmware-builder/

**Comment by Lexa2414118** [comment]
Guys i have the same trouble with WXKG01LM Button. Does firmware update help with this issue?

**Reply by d-allin** [reply]
You can review the [other comment thread](https://github.com/Koenkk/zigbee2mqtt/discussions/15443#discussioncomment-4409143) to find evidence that it does. I have included a full set of instructions there.

**Comment by krzbor** [comment]
I also have a problem with the WXKG01LM Button. After changing to the 655 firmware, the button works properly, but I have problems with the range (low signal quality). It looks like the coordinator is working with less power. One of my sockets is also disconnecting. You can no longer download version 679 to try it out. I tried version 6.7.10 - good signal strength, but no line between WXKG01LM and the coordinator and the button does not work after some time. I don't know what version to download to the device anymore.

**Comment by krzbor** [comment]
I found a solution - if I connect the button to the router and not to the coordinator, everything works fine. My router is a BSD29 socket.

**Reply by litecross91** [reply]
How do this? How can choose where to connect the button?

**Reply by krzbor** [reply]
Normally you have "Permit join (All)", but you can choose another device (router)

**Reply by litecross91** [reply]
I have only one usb Sonoff 3.0 dongle to my rpi. How can I add other routers?

**Reply by krzbor** [reply]
Almost every network-powered Zigbee device is a router that can act as an intermediary between the WXKG01LM and the coordinator.

**Reply by litecross91** [reply]
I see, thank you!

**Comment by mbakgun** [comment]
Hello everyone. If anyone has faced a similar situation, they can follow the steps I took, as they worked smoothly for me. My dongle (coordinator) came with version 6.10.3.  And the same problem happened to me. 
To fix I tried ZHA and Zigbee2MQTT, but my Zigbee radio switch couldn't retain after 45 mins on average, even when the link quality was over 180. 
What I did; Initially, I attempted to update the version to 7.2.3, but I encountered some interesting out-of-memory issues on these applications' boot phase. From that latest version, I decided to downgrade to 679. Now, it's been working like a charm since last night, with no drop-offs.

**Reply by litecross91** [reply]
I don't find 679 FW on this site: https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP
Where can we download it? Thanks!

**Reply by mbakgun** [reply]
Welcome, answered in another thread.

**Reply by sbakopanos** [reply]
What other thread? I've looked and I can't find it...

**Reply by litecross91** [reply]
https://github.com/Koenkk/zigbee2mqtt/discussions/15443#discussioncomment-7301773

**Comment by litecross91** [comment]
Some of my devices are not responding after 1-2 hours (WXKG01LM, RTCGQ01LM) Server: rpi3, HA, Coordinator: dongle-e, FW6.10.3.0 build 297, ezsp v8, Zigbee2mqtt 1.33.1, mosquito broker

**Reply by mbakgun** [reply]
Yes you need to downgrade, I found the flash firmware from their GitHub project. Just go through history and you will see there is a package exists.

**Reply by litecross91** [reply]
I flashed now this: [ncp-uart-sw_6.10.7.0_115200.gbl](https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP/ncp-uart-sw_6.10.7.0_115200.gbl)
2 months ago version. I test this version and "I will be back"

**Reply by litecross91** [reply]
News (31/10/23): Since the firmware update, the sensors are stably connected. :))

**Comment by roman-timoshevskii** [comment]
Hi everyone,

Just received and configured ZBDongle-E and faced with the same issue regarding WXKG01LM stability.
Firmware version out of box is 6.10.3.0 build 297.
Can anyone recommend some firmware that works stable as a lot of newer version released since the last post in this thread.

Thanks in advance.

**Reply by litecross91** [reply]
https://github.com/Koenkk/zigbee2mqtt/discussions/15443#discussioncomment-7301193
I flashed 6.10.7.0 yesterday and it since working :))

**Comment by krzbor** [comment]
I wanted to confirm that firmware 6.10.7 works stably with Xiaomi WXKG01LM. The connection directly to the coordinator is not disconnected.

**Comment by pcmunro** [comment]
After flashing the new firmware I haven't had an issue. The connection would drop previously. This fella does a good job of a walkthrough https://www.youtube.com/watch?v=4eYnURcDrWw
